src/changelog.ts

```diff
@@ -264,28 +354,42 @@ export default class Changelog {
 
   #formatter: Formatter;
 
+  readonly #versionBeforePkgRename: string | undefined;
+
+  readonly #tagPrefixBeforePkgRename: string | undefined;
```

Comment:
Consistent typing for optional properties. 
```suggestion
  readonly #versionBeforePkgRename?: string;

  readonly #tagPrefixBeforePkgRename?: string;
```

Reply:
Yes, I like `| undefined` better than `?:` here, since `undefined` is explicitly being passed in with these two variables, and not making them optional will prevent the code from breaking if `exactOptionalPropertyTypes` is turned on. 

My concern is around consistent typing for the other properties in these files that are defined as optional. `exactOptionalPropertyTypes` would enforce that consistency.